src/coreclr/src/vm/processdiagnosticsprotocolhelper.cpp

```diff
@@ -108,12 +115,68 @@ bool ProcessInfoPayload::Flatten(BYTE * &lpBuffer, uint16_t &cbSize)
     if (fSuccess)
         fSuccess &= TryWriteString(lpBuffer, cbSize, Arch);
 
+    // Array<LPCWSTR>
+    memcpy(lpBuffer, &_nEnvEntries, sizeof(_nEnvEntries));
+    lpBuffer += sizeof(_nEnvEntries);
+    cbSize -= sizeof(_nEnvEntries);
+
+    LPCWSTR cursor = Environment;
+    for (uint32_t i = 0; i < _nEnvEntries; i++)
+    {
+        if (!fSuccess)
+            break;
+
+        uint32_t len = static_cast<uint32_t>(wcslen(cursor) + 1);
+        fSuccess &= TryWriteString(lpBuffer, cbSize, cursor);
+        cursor += len;
+    }
+
     // Assert we've used the whole buffer we were given
     ASSERT(cbSize == 0);
 
     return fSuccess;
 }
 
+void ProcessInfoPayload::EnsureEnv()
```

Comment:
Not sure if this is a good name. Can we rename it to something like ProcessInfoPayload::GetEnvironmentBlock() or ProcessInfoPayload::PopulateEnvBlock()?